src/rpc/net.cpp

```diff
@@ -862,23 +863,20 @@ static RPCHelpMan getnodeaddresses()
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
     }
 
-    int count = 1;
-    if (!request.params[0].isNull()) {
-        count = request.params[0].get_int();
-        if (count < 0) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "Address count out of range");
-        }
-    }
+    const int count{request.params[0].isNull() ? 1 : request.params[0].get_int()};
+    if (count < 0) throw JSONRPCError(RPC_INVALID_PARAMETER, "Address count out of range");
+
     // returns a shuffled list of CAddress
-    std::vector<CAddress> vAddr = node.connman->GetAddresses(count, /* max_pct */ 0);
+    const std::vector<CAddress> vAddr{node.connman->GetAddresses(count, /* max_pct */ 0)};
     UniValue ret(UniValue::VARR);
 
     for (const CAddress& addr : vAddr) {
         UniValue obj(UniValue::VOBJ);
-        obj.pushKV("time", (int)addr.nTime);
-        obj.pushKV("services", (uint64_t)addr.nServices);
+        obj.pushKV("time", static_cast<int>(addr.nTime));
+        obj.pushKV("services", static_cast<uint64_t>(addr.nServices));
```

Comment:
```suggestion
        obj.pushKV("services", uint64_t{addr.nServices});
```

nit: I keep forgetting if this works for service flags, but C++11 casts are preferred over static_casts ;)


(Obviously don't force push unless you have to for other reasons)

Reply:
That does work.

I wonder why we don't print the same services output(s) as for getpeerinfo and getnetworkinfo?